ludwig/train.py

```diff
@@ -396,6 +400,68 @@ def full_train(
     )
 
 
+def kfold_cross_validate(
+                model_definition,
+                model_definition_file=None,
+                data_csv=None,
+                data_train_csv=None,
+                data_validation_csv=None,
+                output_directory='results',
+                k_fold=None,
+                **kwargs
+):
+    logger = logging.getLogger('ludwig')
+    logger.info('starting {:d}-fold cross validation'.format(k_fold))
+
+    # create output_directory if not available
+    if not os.path.isdir(output_directory):
+        os.mkdir(output_directory)
+
+    # read in data to split for the folds
+    data_df = pd.read_csv(data_train_csv)
+
+    # place each fold in a separate directory
+    data_dir = os.path.dirname(data_train_csv)
+    kfold_training_stats = {}
+    for train_index, test_index, fold_num in generate_kfold_splits(data_train_csv, k_fold):
+        with tempfile.TemporaryDirectory(dir=data_dir) as temp_dir_name:
+            # save training and validation subset for the fold into a temporary directory
+            train_csv_fp = os.path.join(temp_dir_name, 'train_fold.csv')
+            test_csv_fp = os.path.join(temp_dir_name, 'test_fold.csv')
+            logger.info("\n\n>>>>> for fold {:d} created temporary directory: {}".format(fold_num, temp_dir_name))
+            data_df.iloc[train_index].to_csv(train_csv_fp, index=False)
+            data_df.iloc[test_index].to_csv(test_csv_fp, index=False)
+
+            # train and validate model on this fold
+            logger.info("training on fold {:d}".format(fold_num))
+            (model,
+             preprocessed_data,
+             _,
+             train_stats,
+             model_definition) = full_train({},
```

Comment:
the {} should not be needed

Reply:
removed `{}`

Reply:
Looks like I was a bit pre-mature in calling this resolved.

There may be an issue in removing `{}` in line 441.  The function definition for `full_train()` looks like this:
```
def full_train(
        model_definition,
        model_definition_file=None,
        data_df=None,
        data_train_df=None,
```

When I remove `{}` from the call to `full_train()` an error is raised.

It appears to remove `{}` from line 441, we have to modify `full_train()`.  

Do you want me to take a stab at modifying `full_train()` as well?

Reply:
Ignore my earlier comment.  I just realized what you were asking me to do. This issues was resolved by this coding sequence:
```
           # train and validate model on this fold
            with open(model_definition_file, 'r') as def_file:
                model_definition = merge_with_defaults(yaml.safe_load(def_file))
            logger.info("training on fold {:d}".format(fold_num))
            (model,
             preprocessed_data,
             _,
             train_stats,
             model_definition) = full_train(
                model_definition,
                data_train_csv=train_csv_fp,
                data_test_csv=test_csv_fp,
                experiment_name='cross_validation',
```

Reply:
You are right, I added that check from API also in full_train: https://github.com/uber/ludwig/commit/7826a1b4f594cf760d5403929fa1ab41302b53de
now you can pass only the file path and it will work.